lib/Gitlab/Client.php

```diff
@@ -250,50 +243,34 @@ public function clearHeaders()
      */
     public function setHeaders(array $headers)
     {
-        $this->httpClient->setHeaders($headers);
+        $this->httpClientBuilder->clearHeaders();
+        $this->httpClientBuilder->addHeaders($headers);
 
         return $this;
     }
 
     /**
-     * @param string $name
-     *
-     * @return mixed
-     *
-     * @throws InvalidArgumentException
+     * @param string $api
+     * @return AbstractApi
      */
-    public function getOption($name)
+    public function __get($api)
```

Comment:
Humm. This should not be in the diff. I will give it a look. Moreover, I also prefer writing methods and remove the magic method and huge comment in the top of the file. It's the same amount of work!